src/typechecker.rs

```diff
@@ -298,6 +349,79 @@ impl<'a> Typechecker<'a> {
                 block,
             } => self.typecheck_def(name, params, return_ty, block, node_id),
             AstNode::Call { ref parts } => self.typecheck_call(parts, node_id),
+            AstNode::For {
+                variable,
+                range,
+                block,
+            } => {
+                self.typecheck_node(range);
+
+                // We don't need to typecheck variable after this
```

Comment:
Why not? (The comment is not very descriptive.)

Reply:
My bad. Basically the variable's is resolved when we typecheck the list. I think the comment is positioned wrong.

Reply:
Not to mention, the variable's type is being set by us after checking the type of the range. We're setting the type in the for node itself, not before.